docs/getting-started-guide.md

```diff
@@ -0,0 +1,168 @@
+CXXFLAGS=-std=c++11
+SYSTEMC_INCLUDE=${HOME}/cosim/systemc-2.3.2/src
+SYSTEMC_LIBDIR=${HOME}/cosim/systemc-2.3.2/src/.libs
+HAVE_VERILOG=n
+HAVE_VERILOG_VERILATOR=n
+HAVE_VERILOG_VCS=n
+VM_TRACE=1
+EOF*
+
+# Build the demo
+make
+```
+
+### Buildroot Setup
+```bash
+cd ~/cosim
+git clone https://github.com/buildroot/buildroot.git
+cd buildroot
+git checkout 36edacce9c2c3b90f9bb11
+mkdir handles
+```
+
+For this walkthough we will be configuring the demo to emulate the Zynq-7000 SoC. The `.dtsi` files available in the Xilinx repositories need significat modification to operate.
+```bash
+# Pull the .dtsi files for the Zynq-7000
+wget https://raw.githubusercontent.com/Xilinx/qemu-devicetrees/master/zynq-pl-remoteport.dtsi
+wget https://raw.githubusercontent.com/Xilinx/linux-xlnx/master/arch/arm/boot/dts/zynq-7000.dtsi
+wget https://raw.githubusercontent.com/Xilinx/device-tree-xlnx/master/device_tree/data/kernel_dtsi/2017.3/BOARD/zc702.dtsi
+
+# Setup the Buildroot compilation configeration
+rm .config .config.old  -f
+cat << EOF | tee ".config"
+BR2_arm=y
+BR2_cortex_a9=y
+BR2_ARM_ENABLE_NEON=y
+BR2_ARM_ENABLE_VFP=y
+BR2_TOOLCHAIN_EXTERNAL=y
+BR2_TOOLCHAIN_EXTERNAL_BOOTLIN=y
+BR2_TOOLCHAIN_EXTERNAL_BOOTLIN_ARMV7_EABIHF_GLIBC_STABLE=y
+BR2_TARGET_GENERIC_GETTY_PORT="ttyPS0"
+BR2_LINUX_KERNEL=y
+BR2_LINUX_KERNEL_CUSTOM_VERSION=y
+BR2_LINUX_KERNEL_CUSTOM_VERSION_VALUE="5.4.75"
+BR2_LINUX_KERNEL_USE_ARCH_DEFAULT_CONFIG=y
+BR2_LINUX_KERNEL_UIMAGE=y
+BR2_LINUX_KERNEL_UIMAGE_LOADADDR="0x8000"
+BR2_LINUX_KERNEL_DTS_SUPPORT=y
+BR2_LINUX_KERNEL_CUSTOM_DTS_PATH="zynq-7000.dtsi  zynq-zc702.dts zynq-pl-remoteport.dtsi"
+BR2_TARGET_ROOTFS_CPIO=y
+BR2_TARGET_ROOTFS_CPIO_GZIP=y
+BR2_TARGET_ROOTFS_EXT2=y
+#BR2_TARGET_ROOTFS_TAR is not set
+EOF
+
+# Build
+make clean && make olddefconfig
+./utils/br/make
+```
+
+## Running the Demo
+In order to run the co-simulation demo, you will need two shells open. One for QEMU and the other for SystemC. In the first shell navigate to the `systemctlm-cosim-demo` directory (`cd ~/cosim/systemctlm-cosim-demo`). This command will start the SystemC simulation:
+```bash
+LD_LIBRARY_PATH=~/cosim/systemc-2.3.2/src/.libs/ ./zynq_demo 	unix:${HOME}/cosim/buildroot/handles/qemu-rport-_cosim@0 1000000
+```
+
+Start the QEMU instance in the second shell by running (in any directory):
+```bash
+${HOME}/cosim/qemu/aarch64-softmmu/qemu-system-aarch64 -M arm-generic-fdt-7series -m 1G -kernel ${HOME}/cosim/buildroot/output/images/uImage -dtb ${HOME}/cosim/buildroot/output/images/zynq-zc702.dtb --initrd ${HOME}/cosim/buildroot/output/images/rootfs.cpio.gz -serial /dev/null -serial mon:stdio -display none -net nic -net nic -net user -machine-path ${HOME}/cosim/buildroot/handles -icount 0,sleep=off -rtc clock=vm -sync-quantum 1000000
+```
+
+## Example Output
+Below is some example output showing the Buildroot instance quererying the SystemC simulated evironment:
+
+```bash
+Welcome to Buildroot
+buildroot login: root
+# devmem 0x40000000
```

Comment:
Maybe describe above what these offsets represent and what you're observing?  i.e., is it a counter?  Additionally, it might be good to add an example of an offset you could write/read back from.

Reply:
This is a good note. We currently don't really describe these _magic numbers_. If I remember correctly, to write we might need to make further additions but this would be a good addition. I believe we were going for documentation as is, then looking to maybe expand.